src/sentry/snuba/discover.py

```diff
@@ -965,7 +967,7 @@ def histogram_query(
         conditions.append([histogram_alias, "<=", max_bin])
 
     columns = [] if key_column is None else [key_column]
-    limit = len(fields) * num_buckets
+    limit = len(fields) * num_buckets * histogram_rows
```

Comment:
would something like 
```suggestion
    groups = len(fields) if histogram_rows is None else histogram_rows
    limit = groups * num_buckets
```
and changing to default to 0 give you better control over the limit?

Reply:
Works for me